### PR TITLE
[Rust] Add support for base64-encoded byte arrays

### PR DESCRIPTION
**Pull request by whirm**
This PR adds support for base64-encoded binary data (type:string, format: byte) for Rust.

- [X] Will commit the complete set of updates samples once the PR is finished and ready to merge to avoid unnecessary noise.

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@frol, @farcaller, @richardwhiuk, @paladinzh, @jacob-pro: Ping! :)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/rust/model.mustache [hunk 1]
```diff
@@ -85,23 +90,27 @@ impl Default for {{classname}} {
 {{!-- for non-enum schemas --}}
 {{^isEnum}}
 {{^discriminator}}
+#[serde_as]
```
Comment (whirm) [inline modules/openapi-generator/src/main/resources/rust/model.mustache:93]: I need some help with this: I don't quite know how to only add this macro if one of the struct attrs. is a byte array.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java [hunk 2]
```diff
@@ -537,6 +537,11 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         if (property.isNullable && !property.required) {
             additionalProperties.put("serdeWith", true);
         }
+
+        // If a property is a base64-encoded byte array, use `serde_with` for deserialization.
+        if (property.isByteArray) {
```
Comment (whirm) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java:542]: Please let me know if you prefer this to be merged with the conditional above.

File: modules/openapi-generator/src/main/resources/rust/model.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/rust/model.mustache:93]: i would do it in post process model: https://github.com/OpenAPITools/openapi-generator/blob/73f1f29eeb19640439fbbd46f19f93dee8d0f91a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java#L280

and add an extension to the model (e.g. x-rust-has-byte-array) if at least one of the properties is a byte array

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java:542]: to me both ways are fine at the first glance

**Comment by whirm** [comment]
@frol, @farcaller, @richardwhiuk, @paladinzh, @jacob-pro, @wing328: I'd say this is ready to merge!

**Comment by wing328** [comment]
@whirm thanks for the PR, which has been merged.

Can you please PM me via Slack when you've time?

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Comment by abdolence** [comment]
This change broke existing code unfortunately.

The culprit is that there are 2 different flavours of Base64: classic and URL Safe.

Now I have Google API that responds with Base64 URL safe which apparently is not supported by serde_with (at least with the current config).

```
Serde(
        Error("Invalid symbol 45, offset 0.", line: 75, column: 37),
    ),
```
Is there any way to disable this feature?

**Comment by whirm** [comment]
@abdolence It is supported by `serde_with`: 

```rust
#[serde_as(as = "Base64<UrlSafe>")]
```

I don't know if openapi specs allow one to specify the encoding character set for base64 encoded bytes though.

Another option would be to add a codegen config option that allows one to choose the character set for all instances.

If you need a quick fix or are not able/willing to cook up a PR, you can use the `templateDir` config option to point to a copy of the original templates modified to use the character set you need.

**Pull request merged**